account_statement_import_fr_cfonb/wizard/account_statement_import.py

```diff
@@ -0,0 +1,206 @@
+        lines = self.split_lines(data_file_decoded)
+        i = 0
+        seq = 1
+        bank_code = guichet_code = account_number = currency_code = False
+        decimals = start_balance = False
+        start_balance = end_balance = False
+        transactions = []
+        for line in lines:
+            i += 1
+            _logger.debug("Line %d: %s" % (i, line))
+            if not line:
+                continue
+            if len(line) != CFONB_WIDTH:
+                raise UserError(
+                    _(
+                        "Line %d is %d caracters long. All lines of a "
+                        "CFONB bank statement file must be 120 caracters long."
+                    )
+                    % (i, len(line))
+                )
+            rec_type = line[0:2]
+            line_bank_code = line[2:7]
+            line_guichet_code = line[11:16]
+            line_account_number = line[21:32]
+            # Some LCL files are invalid: they leave decimals and
+            # currency fields empty on lines that start with '01' and '07',
+            # so I give default values in the code for those fields
+            line_currency_code = line[16:19] != "   " and line[16:19] or "EUR"
+            decimals = line[19:20] != " " and int(line[19:20]) or 2
+            date_cfonb_str = line[34:40]
+            date_dt = False
+            if date_cfonb_str != "      ":
+                date_dt = datetime.strptime(date_cfonb_str, "%d%m%y")
+            if line_account_number in self._excluded_accounts:
+                continue
+            assert decimals == 2, "We use 2 decimals in France!"
+
+            if rec_type == "01":
+                bank_code = line_bank_code
+                guichet_code = line_guichet_code
+                currency_code = line_currency_code
+                account_number = line_account_number
+                transactions = []
+                start_balance = self._parse_cfonb_amount(line[90:104], decimals)
+
+            elif rec_type == "07":
+                end_balance = self._parse_cfonb_amount(line[90:104], decimals)
+                vals_bank_statement = {
+                    "name": account_number,
+                    "date": date_dt,
+                    "balance_start": start_balance,
+                    "balance_end_real": end_balance,
+                    "transactions": transactions,
+                }
+                result.append((currency_code, account_number, [vals_bank_statement]))
+
+            elif rec_type == "04":
+                amount = self._parse_cfonb_amount(line[90:104], decimals)
+                ref = line[81:88].strip()  # This is not unique
+                name = line[48:79].strip()
+                transactions.append(
+                    {
+                        "sequence": seq,
+                        "date": date_dt,
+                        "payment_ref": name,
+                        "unique_import_id": "{}-{}-{:.2f}-{}".format(
```

Comment:
attention il peut avoir deux transactions avec le même `unique_import_id` (cela arrive avec des billets d'avion)

Reply:
Yes, it's not new, this problem has always existed. What do you propose ? Remove unique_import_id completely ? That would disable the auto-drop of already imported lines. My idea: we could make it a configuration option on res.company. What do you think ?

Reply:
```
def _cfonb_unify_transaction(self, transactions):
        """ If case there two operation in the same day, with same amount
            and same ref. """
        unique_import_ids = {}
        for transaction in transactions:
            unique_import_id = transaction['unique_import_id']
            if unique_import_id in unique_import_ids.keys():
                transaction['unique_import_id'] += '-%s' % unique_import_ids[unique_import_id]
                unique_import_ids[unique_import_id] += 1
            else:
                unique_import_ids[unique_import_id] = 1
        return transactions
```

Reply:
OK, I integrated your suggestion, I think it's the least worse solution.